### PR TITLE
feat(providers): repair double-encoded structured output from LLMs

### PR DESCRIPTION
**Pull request by oshorefueled**
## Why

When reviewing content against complex schemas, some LLMs (observed with Anthropic models via Bedrock) emit nested fields — like `violations` — as stringified JSON with unescaped internal quotes. The Vercel AI SDK's structured output parser throws `NoObjectGeneratedError`, and VectorLint surfaces this as an unrecoverable crash. Reviews against affected models cannot complete.

## What this PR covers

- Structured output calls that previously crashed on double-encoded/unescaped-JSON model responses now repair the output and continue reviewing.
- Both structured output entry points (`runPromptStructured` and `runWithTools`) recover from `NoObjectGeneratedError` with identical repair logic.

## Scope

**In scope**
- `src/providers/vercel-ai-provider.ts` — repair helpers, shared `handleNoObjectGenerated` method, wiring in both catch blocks.
- `tests/vercel-ai-provider-repair.test.ts` — 6 tests covering repair paths and fail-safe edge cases.

**Out of scope**
- Prompt-level mitigation (schema simplification, format instructions).
- SDK-native `repairText` integration (requires investigation against SDK version compatibility).

## Behavior Impact

- **User-facing changes:** Yes — reviews that previously crashed with `NoObjectGeneratedError` on affected models now complete successfully. No configuration changes required.
- **Breaking changes:** No.
- **Operational impact:** None — repair runs transparently in the provider layer.

## Risk and Mitigations

- **Risk level:** Low
- **Primary risks:**
  - The state-machine quote-escaping could misclassify a real closing quote as an internal quote on unusual payloads, producing incorrect data.
  - Repair masks a model quality issue that might be better addressed at the prompt or schema level.
- **Mitigations:**
  - Repaired output is validated against the Zod schema before returning; unrepairable payloads still throw.
  - Debug-level logging (`[vectorlint] Structured output repaired after NoObjectGeneratedError`) makes every repair observable.
  - Fail-safe edge cases tested: unparseable outer JSON rethrows, trailing backslash inside string values rethrows.
- **Rollback plan:** Revert this commit — the provider returns to throwing on `NoObjectGeneratedError`.

## API / Contract / Schema Changes

- None.

## How to test / verify

**Checks run**
- `npx vitest run` → 353 passed (55 files)
- `npx tsc --noEmit` → clean

**Manual verification**
- Configure a Bedrock provider with an Anthropic model (e.g. `us.anthropic.claude-haiku-4-5-20251001-v1:0`) in `~/.vectorlint/config.toml`.
- Run `npx tsx src/index.ts <test-doc.md> --model-call single` against a document that triggers a complex schema response.
- Before this PR: crashes with `NoObjectGeneratedError`. After: completes review with findings, quality scores, and token usage.

## Future improvements

- Investigate whether the Vercel AI SDK's `repairText` option can replace the custom repair layer.
- Explore prompt/schema-level mitigation to prevent the double-encoding at the source.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of malformed structured JSON, including nested encoding, unescaped quotes, and control characters.
  * Structured-output and tool-calling responses can now be repaired and schema-validated when possible.
  * Usage metrics are preserved during structured-output recovery attempts.
  * Unrecoverable or invalid responses continue to return the standard structured-output error.

### CODE REVIEW

File: src/providers/vercel-ai-provider.ts [hunk 1]
```diff
@@ -17,6 +17,127 @@ import { handleUnknownError } from '../errors';
...
+    return result;
+  }
+  if (typeof value === 'string' && looksLikeJson(value)) {
+    const repaired = repairJsonString(value);
+    if (repaired !== undefined) {
+      return repairNestedJson(repaired);
+    }
+  }
+  return value;
+}
```
Comment (oshorefueled) [inline src/providers/vercel-ai-provider.ts]: Move to a utils file beside the provider.
